File: CoteMultiLib/Model/Animal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Net;
using Microsoft.EntityFrameworkCore;

namespace CoreMultiLib.Model
{
    //[Table("Animals")]
    [Index(nameof(Name), IsUnique = true)]
    public class Animal
    {
        private Food _food = null!;

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [MaxLength(128)] public string? Name { get; set; }
        public int Age { get; set; }

        public virtual List<Club>?         Clubs   { get; set; }
        public virtual ICollection<Grade>? Grades  { get; set; }
        public virtual Job                 Job     { get; set; } = null!;
        public         int?                JobId   { get; set; }
        public virtual Person?             LovedBy { get; set; }
        public virtual Person?             HatedBy { get; set; }

        [BackingField(nameof(_food))]
        public virtual Food Food
        {
            get => _food;
            set => _food = value;
        }

        // Translates to string in db so Include is not needed.
        public IPAddress IpAddress { get; set; } = null!;

        //public JsonDocument? Passport { get; set; }

        public override string ToString()
        {
            return $"Animal : Id = {Id} Name = {Name} IpAddress = {IpAddress}";
        }
    }
}

File: VBLib_Common/Model/Animal.vb

Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Net
Imports Microsoft.EntityFrameworkCore

Namespace Model
    Public Class Animal
        Private _food As Food
        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Id As Integer
        <MaxLength(128)>
        Public Property Name As String
        Public Property Age As Integer
        Public Overridable Property Clubs As List(Of Club)
        Public Overridable Property Grades As ICollection(Of Grade)
        Public Overridable Property Job As Job
        Public Property JobId As Integer?
        Public Overridable Property LovedBy As Person
        Public Overridable Property HatedBy As Person

        <BackingField(NameOf(_food))>
        Public Overridable Property Food As Food
            Get
                Return _food
            End Get
            Set(ByVal value As Food)
                _food = value
            End Set
        End Property

        Public Property IpAddress As IPAddress

        Public Overrides Function ToString() As String
            Return $"Animal : Id = {Id} Name = {Name} IpAddress = {IpAddress}"
        End Function
    End Class
End Namespace
